Cherrypick 201704

fspiga opened the pull request:
As per Paolo and Pietro request, some changes applied to q-e_schrodinger cherrypicked from SVN commits. Please verify before rebase or squash merge.

fspiga committed:
Cherrypick r133666

fspiga committed:
Cherrypick r13411

fspiga committed:
Cherrypick r13434

fspiga committed:
Cherrypick r13438

fspiga committed:
Cherrypick r13439

fspiga committed:
Cherrypick r13444

fspiga committed:
Cherrypick r13445

fspiga committed:
Cherrypick r13448

fspiga committed:
Cherrypick r13449

fspiga committed:
Fix cherrypick r13444, missing tau_units

giannozz commented:
r133666, r13444 and its fix contain changes that shouldn't go into the stable version. All others seem ok to me. r13455 (limited to PW/src/memory_report.f90) should also be merged.

fspiga commented:
I'll get r13455 inside but please fix the few things that should not go inside this version of q-e_schrodinger

Commit:
1. setup.f90 reverted to safe version
2. improved version of memory_report.f90
3. tentative fix for exact exchange with noncolinear magnetism
4. minor bug in vdW kernel file generatiom
5. a few erroneous error messages
6. fix for a function used only with "environ"
